tests/functional/ArchiveTest.php

```diff
@@ -0,0 +1,181 @@
+<?php
+
+namespace Unish;
+
+use Webmozart\PathUtil\Path;
+
+/**
+ * Class ArchiveTest.
+ *
+ * @group slow
+ * @group commands
+ * @group archive
+ */
+class ArchiveTest extends CommandUnishTestCase
+{
+    /**
+     * @var string
+     */
+    protected string $archivePath;
+
+    /**
+     * @var array
+     */
+    protected array $archiveDumpOptions;
+
+    /**
+     * @inheritdoc
+     */
+    public function setUp(): void
+    {
+        $this->setUpDrupal(1, true);
+        $this->archiveDumpOptions = [
+            'db' => null,
+            'files' => null,
+            'code' => null,
+            'exclude-code-paths' => 'sut/sites/.+/settings.php,(?!sut).*',
+        ];
+
+        $this->archivePath = Path::join($this->getSandbox(), 'archive.tar.gz');
+        $this->drush(
+            'archive:dump',
+            [],
+            array_merge($this->archiveDumpOptions, [
+                'destination' => $this->archivePath,
+                'overwrite' => null,
+            ])
+        );
+        $actualArchivePath = $this->getOutput();
+        $this->assertEquals($this->archivePath, $actualArchivePath);
+    }
+
+    public function testArchiveDumpCommand(): void
+    {
+        // Create an archive.
+        $this->drush(
+            'archive:dump',
+            [],
+            $this->archiveDumpOptions
+        );
+        $actualArchivePath = $this->getOutput();
+        $this->assertMatchesRegularExpression(
+            '#\/archives\/\d+\/archive\.tar\.gz$#',
+            $actualArchivePath
+        );
+
+        // Try to overwrite the existing archive with "--destination".
+        $this->drush(
+            'archive:dump',
+            [],
+            array_merge($this->archiveDumpOptions, [
+                'destination' => $this->archivePath,
+            ]),
+            null,
+            null,
+            self::EXIT_ERROR
+        );
+        $this->assertStringContainsString('The destination file already exists.', $this->getErrorOutput());
+
+        // Overwrite the existing archive with "--destination" and "--override".
+        $this->drush(
+            'archive:dump',
+            [],
+            array_merge($this->archiveDumpOptions, [
+                'destination' => $this->archivePath,
+                'overwrite' => null,
+            ])
+        );
+        $actualArchivePath = $this->getOutput();
+        $this->assertEquals($this->archivePath, $actualArchivePath);
+
+        // Validate database credentials are present in settings.php file.
```

Comment:
I think the intention here is to test the `overwrite` option.

Reply:
I reworked the test to use `file_put_contents` instead of the first `archive:dump` so that we can still test `--overwrite` and not call dump so many times.